AppSyncRealTimeClient/ConnectionProvider/AppsyncRealtimeConnection/RealtimeConnectionProviderResponse.swift

```diff
@@ -52,3 +52,36 @@ extension RealtimeConnectionProviderResponse: Decodable {
         case responseType = "type"
     }
 }
+
+extension RealtimeConnectionProviderResponse {
+
+    func isMaxSubscriptionReached() -> Bool {
+
+        guard let payload = payload else {
+            return false
```

Comment:
It's not immediately clear why `payload` being `nil` should cause this to return `false`.
I think documenting the reason here, and/or adding a testcase asserting `RealtimeConnectionProviderResponse(payload: nil).isMaxSubscriptionReached() == false` would be helpful when looking back at this in the future.

Reply:
Added some more documentation around this and unit tests. The main reason why `payload` is nil is because the response comes back in different shapes according to the type: https://docs.aws.amazon.com/appsync/latest/devguide/real-time-websocket-client.html#error-message